### ISSUE TITLE
Don't call `rewind` on f

### ISSUE THREAD

**Issue by KevinColemanInc** [tilo/smarter_csv#142]
If you pass in a file with a bom character, the rewind jumps back to before the bom.

```
file = File.open(csv_file, "r:bom|utf-8")
file.readline.first != "\xEF" # true - no BOM character
file.rewind
file.readline.first == "\xEF" # true - BOM character
```

This is a problem for required and missing headers, because the first header value will contain the blank character and not correctly match the given string.

Problem line:
[1.2-stable/lib/smarter_csv/smarter_csv.rb#L35](https://github.com/tilo/smarter_csv/blob/1.2-stable/lib/smarter_csv/smarter_csv.rb#L35)

Related stack overflow:
[https://stackoverflow.com/a/7780531](https://stackoverflow.com/a/7780531)

**Comment by tilo** [comment]
are you passing a string instead of the options hash? 

did you mean: `file = File.open(csv_file, encoding: "r:bom|utf-8")`

can you provide a sample file?

**Comment by mohanwitz** [comment]
@tilo 
This problem is still there, any updates on when it will get fixed?

**Comment by tilo** [comment]
https://github.com/tilo/smarter_csv/blob/579c0715e57f12c913f820ac2ae117e71b4f196d/lib/smarter_csv/smarter_csv.rb#L35

**Comment by tilo** [comment]
@mohanwitz can you try this:

```
data = SmarterCSV.process(your_filename, file_encoding: 'bom|utf-8')
```

**Comment by tilo** [comment]
BOM issue was fixed in version 1.8.0
